src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -757,6 +757,30 @@ public static string GetLongDescriptionAnnotation(this IEdmModel model, IEdmVoca
             return null;
         }
 
+        /// <summary>
+        /// Gets the collection of string for term Org.OData.Validation.V1.DerivedTypeConstraint from a target annotatable
+        /// </summary>
+        /// <param name="model">The model referenced to.</param>
+        /// <param name="target">The target Annotatable to find annotation</param>
+        /// <returns>Null or a collection string of qualifed type name</returns>
+        public static IList<string> GetDerivedTypeConstraintAnnotation(this IEdmModel model, IEdmVocabularyAnnotatable target)
+        {
+            EdmUtil.CheckArgumentNull(model, "model");
+            EdmUtil.CheckArgumentNull(target, "target");
+
+            IEdmVocabularyAnnotation annotation = model.FindVocabularyAnnotations<IEdmVocabularyAnnotation>(target, ValidationVocabularyModel.DerivedTypeConstraintTerm).FirstOrDefault();
+            if (annotation != null)
+            {
+                IEdmCollectionExpression collectionExpression = annotation.Value as IEdmCollectionExpression;
+                if (collectionExpression != null && collectionExpression.Elements != null)

```

Comment:
>collectionExpression.Elements != null [](start = 52, length = 37)

It should be legal to have the annotation with an empty (or omitted?) list of derived types, in which case the type of the element must be exactly the expected type (no derived types allowed).  Can we add a test for this case?